Uninitialized field, could cause invalid timeouts to be specified to ldap_bind,
resulting in EINVAL and termination of sendmail during initialization of LDAP
maps.

Also, fix a debug printout of a time_t field on ILP32 architectures where the
64-bit entity ws casted to long, instead of long long.

Index: libsm/ldap.c
--- libsm/ldap.c.orig
+++ libsm/ldap.c
@@ -79,6 +79,9 @@ sm_ldap_clear(lmap)
 	lmap->ldap_options = 0;
 # endif
 	lmap->ldap_attrsep = '\0';
+#  if LDAP_NETWORK_TIMEOUT
+	lmap->ldap_networktmo = 0;
+#  endif
 	lmap->ldap_binddn = NULL;
 	lmap->ldap_secret = NULL;
 	lmap->ldap_method = LDAP_AUTH_SIMPLE;
@@ -439,9 +442,9 @@ sm_ldap_start(name, lmap)
 			lmap->ldap_method);
 	save_errno = errno;
 	if (sm_debug_active(&SmLDAPTrace, 9))
-		sm_dprintf("ldap_bind(%s)=%d, errno=%d, tmo=%ld\n",
+		sm_dprintf("ldap_bind(%s)=%d, errno=%d, tmo=%lld\n",
 			lmap->ldap_uri, msgid, save_errno,
-			(long) tmo.tv_sec);
+			(long long) tmo.tv_sec);
 	if (-1 == msgid)
 	{
 		r = -1;
